From<&str, Arc<dyn arrow2::array::Array> for Series not working with masters polars banch

Getting the following errors when trying to compile the code  against the master branch of polars. Might be something we want to get ahead of?

```
  Compiling geopolars v0.1.0 (/home/stuart/Projects/geopolars/geopolars)
error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:131:9
    |
131 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow3::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:131:9
    |
115 |     fn centroid(&self) -> Result<Series> {
    |                           -------------- expected `Result<Series, PolarsError>` because of return type
...
131 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:131:9
    |
131 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow3::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:156:9
    |
156 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:156:9
    |
134 |     fn convex_hull(&self) -> Result<Series> {
    |                              -------------- expected `Result<Series, PolarsError>` because of return type
...
156 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:156:9
    |
156 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:175:9
    |
175 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:175:9
    |
159 |     fn envelope(&self) -> Result<Series> {
    |                           -------------- expected `Result<Series, PolarsError>` because of return type
...
175 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:175:9
    |
175 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:210:9
    |
210 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:210:9
    |
178 |     fn euclidean_length(&self) -> Result<Series> {
    |                                   -------------- expected `Result<Series, PolarsError>` because of return type
...
210 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:210:9
    |
210 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:229:9
    |
229 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:229:9
    |
213 |     fn exterior(&self) -> Result<Series> {
    |                           -------------- expected `Result<Series, PolarsError>` because of return type
...
229 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:229:9
    |
229 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:245:22
    |
245 |         let series = Series::try_from(("geometry", Arc::new(array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:245:22
    |
245 |         let series = Series::try_from(("geometry", Arc::new(array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:363:9
    |
363 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:363:9
    |
249 |     fn geodesic_length(&self, method: GeodesicLengthMethod) -> Result<Series> {
    |                                                                -------------- expected `Result<Series, PolarsError>` because of return type
...
363 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:363:9
    |
363 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:387:9
    |
387 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:387:9
    |
366 |     fn geom_type(&self) -> Result<Series> {
    |                            -------------- expected `Result<Series, PolarsError>` because of return type
...
387 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:387:9
    |
387 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:400:9
    |
400 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:400:9
    |
390 |     fn is_empty(&self) -> Result<Series> {
    |                           -------------- expected `Result<Series, PolarsError>` because of return type
...
400 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:400:9
    |
400 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:416:9
    |
416 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:416:9
    |
403 |     fn is_ring(&self) -> Result<Series> {
    |                          -------------- expected `Result<Series, PolarsError>` because of return type
...
416 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:416:9
    |
416 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:444:9
    |
444 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:444:9
    |
419 |     fn simplify(&self, tolerance: f64) -> Result<Series> {
    |                                           -------------- expected `Result<Series, PolarsError>` because of return type
...
444 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:444:9
    |
444 |         Series::try_from(("geometry", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:463:9
    |
463 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:463:9
    |
447 |     fn x(&self) -> Result<Series> {
    |                    -------------- expected `Result<Series, PolarsError>` because of return type
...
463 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:463:9
    |
463 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:482:9
    |
482 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0308]: mismatched types
   --> geopolars/src/geoseries.rs:482:9
    |
466 |     fn y(&self) -> Result<Series> {
    |                    -------------- expected `Result<Series, PolarsError>` because of return type
...
482 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected enum `PolarsError`, found enum `Infallible`
    |
    = note: expected enum `Result<_, PolarsError>`
               found enum `Result<_, Infallible>`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:482:9
    |
482 |         Series::try_from(("result", Arc::new(result) as ArrayRef))
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `geopolars` due to 38 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:533:22
    |
533 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:533:22
    |
533 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:573:22
    |
573 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:573:22
    |
573 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:597:22
    |
597 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:597:22
    |
597 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:626:22
    |
626 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:626:22
    |
626 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:658:22
    |
658 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error[E0277]: the trait bound `Series: From<(&str, Arc<dyn arrow2::array::Array>)>` is not satisfied
   --> geopolars/src/geoseries.rs:658:22
    |
658 |         let series = Series::try_from(("geometry", Arc::new(test_array) as ArrayRef)).unwrap();
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `From<(&str, Arc<dyn arrow2::array::Array>)>` is not implemented for `Series`
    |
    = help: the following other types implement trait `From<T>`:
              <Series as From<ChunkedArray<T>>>
              <Series as From<Logical<DateType, Int32Type>>>
              <Series as From<Logical<DatetimeType, Int64Type>>>
              <Series as From<Logical<DurationType, Int64Type>>>
              <Series as From<Logical<TimeType, Int64Type>>>
    = note: required because of the requirements on the impl of `Into<Series>` for `(&str, Arc<dyn arrow2::array::Array>)`
    = note: required because of the requirements on the impl of `TryFrom<(&str, Arc<dyn arrow2::array::Array>)>` for `Series`

error: could not compile `geopolars` due to 48 previous errors
```

--- a/geopolars/Cargo.toml
+++ b/geopolars/Cargo.toml
@@ -6,8 +6,7 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-arrow2 = { version = "0.11.2", features = ["io_ipc"] }
 geozero = { version = "0.9.4", features = ["with-wkb"] }
 rstar = "0.9.3"
 geo = { git = "https://github.com/georust/geo.git", rev = '578e213875915e1f895892487b5a36ca0d91fac3' }
-polars = { version = "0.21.1", features = ["ipc", "dtype-u8", "dtype-i8"] }
+polars = { version = "0.22.8", features = ["ipc", "dtype-u8", "dtype-i8"] }

--- a/geopolars/src/geoseries.rs
+++ b/geopolars/src/geoseries.rs
@@ -1,13 +1,13 @@
 use std::sync::Arc;
 
 use crate::util::iter_geom;
-use arrow2::array::{
-    ArrayRef, BinaryArray, BooleanArray, MutableBinaryArray, MutableBooleanArray,
-    MutablePrimitiveArray, PrimitiveArray,
-};
 use geo::algorithm::affine_ops::AffineTransform;
 use geo::{map_coords::MapCoords, Geometry, Point};
 use geozero::{CoordDimensions, ToWkb};
+use polars::export::arrow::array::{
+    ArrayRef, BinaryArray, BooleanArray, MutableBinaryArray, MutableBooleanArray,
+    MutablePrimitiveArray, PrimitiveArray,
+};
 use polars::prelude::{PolarsError, Result, Series};
 use std::convert::Into;
 
@@ -677,9 +677,9 @@ mod tests {
     use polars::prelude::Series;
     use std::sync::Arc;
 
-    use arrow2::array::{ArrayRef, BinaryArray, MutableBinaryArray};
     use geo::{line_string, polygon, CoordsIter, Geometry, LineString, MultiPoint, Point};
     use geozero::{CoordDimensions, ToWkb};
+    use polars::export::arrow::array::{ArrayRef, BinaryArray, MutableBinaryArray};
 
     use super::TransformOrigin;
 

